airbyte-integrations/connectors/source-intercom/source_intercom/spec.json

```diff
@@ -4,20 +4,85 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Source Intercom Spec",
     "type": "object",
-    "required": ["access_token", "start_date"],
-    "additionalProperties": false,
+    "required": ["start_date"],
+    "additionalProperties": true,
     "properties": {
-      "access_token": {
-        "type": "string",
-        "description": "Intercom Access Token. See the <a href=\"https://docs.airbyte.io/integrations/sources/intercom\">docs</a> for more information on how to obtain this key.",
-        "airbyte_secret": true
-      },
       "start_date": {
         "type": "string",
         "description": "The date from which you'd like to replicate data for Intercom API, in the format YYYY-MM-DDT00:00:00Z. All data generated after this date will be replicated.",
         "examples": ["2020-11-16T00:00:00Z"],
         "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$"
+      },
+      "authorization": {
+        "type": "object",
+        "title": "Authentication Type",
+        "oneOf": [
+          {
+            "title": "Oauth authentication",
+            "type": "object",
+            "required": [
+              "auth_type",
+              "client_id",
+              "client_secret",
+              "access_token"
+            ],
+            "properties": {
+              "auth_type": {
+                "type": "string",
+                "const": "Oauth",
+                "enum": ["Oauth"],
+                "default": "Oauth",
+                "order": 0
+              },
```

Comment:
if client ID/secret are not used inside the connector then we don't need to include them in the spec. as far as I can tell we can actually keep the old spec format and just annotate using the oauth parameters. WDYT?

Reply:
Hi Sherif,

IDEA: do not change format of spec.json file, only add oauth2 related annotations.

I was not able to implement such idea. Could you help me to understand how you see its realization?

I have tried different combinations but they do not work for me:

{
   <ORIGINAL FORMAT>
   
  "authSpecification": {
    "auth_type": "oauth2.0",
    "oauth2Specification": {
      "rootObject": [],
      "oauthFlowInitParameters": [["client_id"], ["client_secret"]],
      "oauthFlowOutputParameters": [["access_token"]]
    }
  }
}

The problems are:
- Problem1: spec.json validation fails:

    : spec validation:
            diff = params - schema_path
    >       assert diff == set(), f"Specified oauth fields are missed from spec schema: {diff}"
    E       AssertionError: Specified oauth fields are missed from spec schema: {'/client_secret', '/client_id'}
    E       assert equals failed
    E         set([                set()               
    E           '/client_id',                          
    E           '/client_secret',                      
    E         ])

    so it means that oauth2 annotation requires 'client_id'/'client_secret' properties to be declared in spec file.
    SOLUTION would be to add 'client_id'/'client_secret' properties to spec.json
    
- Problem2: user is not able to specify the needed credentials type.
    SOLUTION: add new property which allows customer to choose needed type